analysis/src/code.rs

```diff
@@ -162,22 +184,41 @@ impl CodeAnalyzer {
         Ok(code_reports)
     }
 
+    pub fn get_package_dependencies<'a>(
+        &self,
+        graph: &'a PackageGraph,
+        package: &PackageMetadata,
+    ) -> Result<Vec<PackageMetadata<'a>>> {
+        let dependencies: Vec<PackageMetadata> = graph
+            .query_forward(iter::once(package.id()))?
+            .resolve()
+            .packages(DependencyDirection::Forward)
+            .filter(|pkg| pkg.id() != package.id())
+            .collect();
+        Ok(dependencies)
+    }
```

Comment:
This can be simplified using methods directly on the PackageMetadata itself:

```
pub fn get_package_dependencies<'a>(
        package: &PackageMetadata<'a>,
    ) -> Vec<PackageMetadata<'a>> {
        package.direct_links().map(|link| link.to()).collect() 
    }

```

Reply:
we're pulling all (direct+transitive) deps here.